### ISSUE TITLE
Move OPAM pins to an OPAM repo

### ISSUE THREAD

**Issue by djwillia** [djwillia/solo5#10]
From @avsm on the mirage-devel mailing list regarding Docker packaging:

>instead of individual pins, you can also create a git repository that has a `packages/` subdirectory and has all the pinned packages.  This cluster of pins can be added to your working set via `opam repo add <name> <url>`.  For example, see https://github.com/mirage/mirage-dev

**Comment by mato** [comment]
Look at https://github.com/mato/opam-rumprun for an example of how such a repository is structured.

**Comment by djwillia** [comment]
Thanks.  The tricky part is how the current franken-build system does not respect the boundaries of an OPAM package.  I'm almost done fixing the Makefile.app so that it does, which is a baby step toward a more sane/compatible build process.

**Comment by mato** [comment]
From yesterday's conversation on Slack:
>it sounds like the todo list should be something like:
1. move cstubs from mirage components into platform (to get all of the need for CFLAGS, includes, etc) into one place
2. change the kernel Makefile to produce a .o
3. produce the solo5-kernel-ukvm package
4. properly send the requisite info to compile platform cstubs and link against the .o

**Comment by mato** [comment]
Adding more steps done, for the record (so that we don't forget what we did):
>1. `mirage-solo5` (the OPAM package) needs to gain a dependency on `solo5-kernel-ukvm`.
2. `mirage-solo5` (or some other package) needs to build the stubs I removed from the standalone kernel build (app_undefined.c?).
3. `solo5-kernel-ukvm` needs to be added to `djwillia/opam-solo5` (the Solo5 OPAM repository).
4. The mirage tool in your fork of the `mirage` package needs to be taught about the link step.
